### PR TITLE
fixed passages

### PR DESCRIPTION
**Pull request by rhagarty**
- if query by passages, only keep matches that have a corresponding passage
- fixed sort issue to keep track of internal sort keys and external sort keys (used for Discovery queries)
- update some documentation

### CODE REVIEW

File: DEVELOPING.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ Below is a general overview of the most important directories and files in the p
 ├── data # Contains data files used by this code pattern
 │   └── airbnb
 │       └── *.json # Austin TX Airbnb JSON files
+├── lib # library and utility functions
+│   ├── utils.js # Utility functions/constants used by src and server code
+│   └── watson-discovery-setup.js # Funcions to auto-load data files into Discover
```
Comment (markstur) [inline DEVELOPING.md]: s/Discover/Discovery/

File: server/index.js [hunk 2]
```diff
@@ -195,9 +195,9 @@ function createServer() {
 
         // get all the results data in right format
         var matches = utils.parseData(json);
-        matches = utils.formatData(matches, [], false);
+        matches = utils.formatData(matches, []);
         var totals = utils.getTotals(matches);
-    
+
```
Comment (markstur) [inline server/index.js]: looks like trailing whitespace was introduced

File: server/index.js [hunk 3]
```diff
@@ -238,16 +238,16 @@ function createServer() {
 
           // get all the results data in right format
           var matches = utils.parseData(results);
-          matches = utils.formatData(matches, [], false);
+          matches = utils.formatData(matches, []);
           var totals = utils.getTotals(matches);
-    
+
```
Comment (markstur) [inline server/index.js]: trailing white space

**Comment by rhagarty** [comment]
Thanks Mark. Fixed

**Pull request merged**